SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -214,20 +217,46 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 	return NewInterest;
 }
 
-void InterestFactory::AddUserDefinedQueries(const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+void InterestFactory::AddActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
 {
-	check(Actor);
+	check(InActor);
 	check(ClassInfoManager);
 
 	TArray<UActorInterestComponent*> ActorInterestComponents;
-	Actor->GetComponents<UActorInterestComponent>(ActorInterestComponents);
+	InActor->GetComponents<UActorInterestComponent>(ActorInterestComponents);
 	if (ActorInterestComponents.Num() == 1)
 	{
 		ActorInterestComponents[0]->CreateQueries(*ClassInfoManager, LevelConstraints, OutQueries);
 	}
 	else if (ActorInterestComponents.Num() > 1)
 	{
-		UE_LOG(LogInterestFactory, Error, TEXT("%s has more than one ActorInterestQueryComponent"), *Actor->GetPathName());
+		UE_LOG(LogInterestFactory, Error, TEXT("%s has more than one ActorInterestQueryComponent"), *InActor->GetPathName());
+	}
+}
+
+void InterestFactory::RecursivelyAddUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+{
+	check(InActor);
+
+	AddActorUserDefinedQueries(InActor, LevelConstraints, OutQueries);
+	for (const auto* Child : InActor->Children)
+	{
+		RecursivelyAddUserDefinedQueries(Child, LevelConstraints, OutQueries);
+	}
+}
+
+void InterestFactory::AddUserDefinedQueries(const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+{
+	SCOPE_CYCLE_COUNTER(STAT_InterestFactoryAddUserDefinedQueries);
+
+	if (APlayerController* PlayerController = Cast<APlayerController>(Actor))
+	{
+		RecursivelyAddUserDefinedQueries(Actor, LevelConstraints, OutQueries);
+		RecursivelyAddUserDefinedQueries(PlayerController->GetPawn(), LevelConstraints, OutQueries);
```

Comment:
If Pawn isn't set, we'll crash. Can we safely return from the Recursive add instead?

Reply:
nit: slight code reduction, can just pass the Pawn() if we're a PC, and do the Actor regardless prior to the check.